File: src/main/java/eu/fabiool/cnpj/file/parsing/lib/model/CnaeSecundaria.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package eu.fabiool.cnpj.file.parsing.lib.model;

import static java.util.Arrays.copyOfRange;

/**
 *
 * @author Fabio
 */
public class CnaeSecundaria extends AbstractInfoCnpj {

    private final char tipoDoRegistro;
    private final char indicadorFullDiario;
    private final char tipoAtualização;
    private final String cnpj;
    private final String cnaeSecundaria;
    private final String filler;
    private final char fimDeRegistro;
    
    public static CnaeSecundaria parse(final byte[] source) {
        return new CnaeSecundaria(
                new String(copyOfRange(source, 0, 1)).charAt(0),
                new String(copyOfRange(source, 1, 2)).charAt(0),
                new String(copyOfRange(source, 2, 3)).charAt(0),
                new String(copyOfRange(source, 3, 17)).trim(),
                new String(copyOfRange(source, 17, 488)).trim(),
                new String(copyOfRange(source, 488, 1199)).trim(),
                new String(copyOfRange(source, 1199, 1200)).charAt(0)
        );
    }

    public CnaeSecundaria(char tipoDoRegistro, char indicadorFullDiario, char tipoAtualização, String cnpj, String cnaeSecundaria, String filler, char fimDeRegistro) {
        this.tipoDoRegistro = tipoDoRegistro;
        this.indicadorFullDiario = indicadorFullDiario;
        this.tipoAtualização = tipoAtualização;
        this.cnpj = cnpj;
        this.cnaeSecundaria = cnaeSecundaria;
        this.filler = filler;
        this.fimDeRegistro = fimDeRegistro;
    }

    public char getTipoDoRegistro() {
        return tipoDoRegistro;
    }

    public char getIndicadorFullDiario() {
        return indicadorFullDiario;
    }

    public char getTipoAtualização() {
        return tipoAtualização;
    }

    public String getCnpj() {
        return cnpj;
    }

    public String getCnaeSecundaria() {
        return cnaeSecundaria;
    }

    public String getFiller() {
        return filler;
    }

    public char getFimDeRegistro() {
        return fimDeRegistro;
    }
}

File: src/main/kotlin/model/CnaeSecundaria.kt
package model

import com.sun.xml.internal.ws.spi.db.BindingContextFactory.LOGGER
import java.util.*
import java.util.logging.Level

class CnaeSecundaria constructor(): InfoCnpj {

    var tipoDoRegistro: Char? = null
    var indicadorFullDiario: Char? = null
    var tipoAtualização: Char? = null
    lateinit var cnpj: String
    lateinit var cnaeSecundaria: String
    lateinit var filler: String
    var fimDeRegistro: Char? = null

    fun parse(source: ByteArray): CnaeSecundaria {

        LOGGER.log(Level.INFO, "Trying to parse CnaeSecundaria instance")

        return CnaeSecundaria(
                String(Arrays.copyOfRange(source, 0, 1))[0],
                String(Arrays.copyOfRange(source, 1, 2))[0],
                String(Arrays.copyOfRange(source, 2, 3))[0],
                String(Arrays.copyOfRange(source, 3, 17)),
                String(Arrays.copyOfRange(source, 17, 488)),
                String(Arrays.copyOfRange(source, 488, 1199)),
                String(Arrays.copyOfRange(source, 1199, 1200))[0]
        )
    }

    constructor(tipoDoRegistro: Char, indicadorFullDiario: Char,
                tipoAtualização: Char, cnpj: String,
                cnaeSecundaria: String, filler: String,
                fimDeRegistro: Char) : this() {
        this.tipoDoRegistro = tipoDoRegistro
        this.indicadorFullDiario = indicadorFullDiario
        this.tipoAtualização = tipoAtualização
        this.cnpj = cnpj
        this.cnaeSecundaria = cnaeSecundaria
        this.filler = filler
        this.fimDeRegistro = fimDeRegistro
    }
}